src/source/Signaling/Signaling.c

```diff
@@ -527,6 +528,32 @@ STATUS validateSignalingClientInfo(PSignalingClient pSignalingClient, PSignaling
     // Copy and store internally
     pSignalingClient->clientInfo = *pClientInfo;
 
+    // V1 features
+    switch (pSignalingClient->clientInfo.signalingClientInfo.version) {
+        case 0:
+            // Set the default path
+            STRCPY(pSignalingClient->clientInfo.cacheFilePath, DEFAULT_CACHE_FILE_PATH);
+
+            break;
+
+        case 1:
+            // If the path is specified and not empty then we validate and copy/store
+            if (pSignalingClient->clientInfo.signalingClientInfo.cacheFilePath != NULL &&
+                pSignalingClient->clientInfo.signalingClientInfo.cacheFilePath[0] != '\0') {
+                CHK(STRNLEN(pSignalingClient->clientInfo.signalingClientInfo.cacheFilePath, MAX_PATH_LEN + 1) <= MAX_PATH_LEN,
+                    STATUS_SIGNALING_INVALID_CLIENT_INFO_CACHE_FILE_PATH_LEN);
+                STRCPY(pSignalingClient->clientInfo.cacheFilePath, pSignalingClient->clientInfo.signalingClientInfo.cacheFilePath);
+            } else {
+                // Set the default path
+                STRCPY(pSignalingClient->clientInfo.cacheFilePath, DEFAULT_CACHE_FILE_PATH);
+            }
+
+            break;
+
+        default:
```

Comment:
Just wondering, do we want to fail here? Maybe we can fallback on v0 instead of failing?

Reply:
This really is an internal code failure case - more like an assert. We should never reach this state as we validate the versions but in case we add another version at some stage and forget to add the case for it. I would, in fact, use CHECK to crash the execution but this is similar to just returning a failure